### PR TITLE
Pytest improvements and cleanup

### PR DESCRIPTION
**Pull request by millerdev**
Follow-up to https://github.com/dimagi/commcare-hq/pull/34778

:blowfish: Review by commit.

### Safety story

Only affects tests.

### Automated test coverage

Yes.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/tests/pytest_plugins/reusedb.py [hunk 1]
```diff
@@ -442,4 +442,15 @@ def unblock():
...
 
+class BlockedMock(Mock):
+    def __getattr__(self, name):
+        try:
+            return super().__getattr__(name)
+        except AttributeError:
+            raise RuntimeError(
+                'Database access not allowed, use the "django_db" mark, or '
+                'the "db" or "transactional_db" fixtures to enable it.'
+            )
```
Comment (gherceg) [inline corehq/tests/pytest_plugins/reusedb.py:470]: As mentioned in https://github.com/dimagi/commcare-hq/pull/34778#discussion_r1854551526, I don't think me commenting on this originally was worthwhile. I was mainly pondering if there was a way to get a `RuntimeError` "organically", not to create a mock that translates an AttributeError into a RuntimeError just for the sake of matching the other dbs in the test. I'm open to removing this if you are.

File: custom/inddex/tests/test_master_report.py [hunk 2]
```diff
@@ -97,6 +97,15 @@ def with_db(func):
...
 
+def setup_module():
+    """Make dividedwerun run all tests in this module on the same node
+
+    This could be eliminated if dividedwerun had a way to determine that
+    tests in this module use a fixture (inddex_domain) that uses the
+    database.
+    """
+
+
```
Comment (gherceg) [inline custom/inddex/tests/test_master_report.py:108]: Can you elaborate on how including a `setup_module` function ensures dividewerun runs all tests in this module on the same node?

Comment (millerdev) [reply custom/inddex/tests/test_master_report.py:108]: https://github.com/dimagi/commcare-hq/blob/a908d8eb05f6af4285da97dd5ae58998c0a1ad99/corehq/tests/pytest_plugins/dividedwerun.py#L51-L53

File: corehq/tests/pytest_plugins/reusedb.py (hunk 1 above)
Comment (millerdev) [reply corehq/tests/pytest_plugins/reusedb.py:470]: I do think this error message is better than the attribute error, so it does seem like an improvement. Given that the [other source of this error](https://github.com/pytest-dev/pytest-django/blob/1ffc3239387bcfadbd8aecfdfacf3d6b52b947f8/pytest_django/plugin.py#L832-L836) is `pytest-django`, it seems reasonable for this module to raise a similar error for Couch.

**Review by gherceg — APPROVED** [review]

**Pull request merged**